lib/active_record/virtual_attributes/virtual_fields.rb

```diff
@@ -176,15 +176,22 @@ def without_virtual_includes
       end
     end
 
-    include Module.new {
+    include(Module.new {
       # From ActiveRecord::FinderMethods
       def find_with_associations
         real = without_virtual_includes
         return super if real.equal?(self)
 
-        recs = real.find_with_associations
+        if ActiveRecord.version.to_s >= "5.1"
+          recs, join_dep = real.find_with_associations { |relation, join_dependency| [relation, join_dependency] }
+        else
+          recs = real.find_with_associations
+        end
         MiqPreloader.preload(recs, preload_values + includes_values) if includes_values
 
+        if ActiveRecord.version.to_s >= "5.1"
+          yield recs, join_dep
+        end
```

Comment:
Can these conditionals be removed?  I thought 5.0 and earlier didn't take advantage of them.  Maybe yield if block_given?  I was hoping we can do the logic for 5.1 in a 5.0 compatible way.

Reply:
`block_given?` instead?

Reply:
part of the version check tells us how to proceed further when we deprecate 5.0
So we'll know to remove the extra if check.

While version checking is uglier, it does tell us how to deprecate code for older versions

Reply:
Right, I'm not sure if it's even needed though.  I'm fine if it is.

Reply:
Looks good 👍